src/quality_result_gui/quality_error_manager.py

```diff
@@ -157,6 +162,41 @@ def _add_predefined_filters(self) -> None:
             self._attribute_filter.update_filter_from_errors
         )
 
+    def _add_actions(self) -> None:
+        self.toggle_show_errors_on_map_action = QAction(
+            tr("Toggle show errors on map filter"), self.dock_widget
+        )
+        self.toggle_show_errors_on_map_action.triggered.connect(
+            self._toggle_show_errors_on_map_filter
+        )
+        self._add_shortcut_for_action(
+            self.toggle_show_errors_on_map_action,
+            QualityResultManager.SHORTCUT_TOGGLE_ERRORS_ON_MAP_FILTER,
+        )
+
+    def _add_shortcut_for_action(
+        self,
+        action: QAction,
+        shortcut: str,
+    ) -> None:
+        iface.mainWindow().addAction(action)
+
+        QgsGui.shortcutsManager().registerAction(action, shortcut)
+        action.setShortcut(QKeySequence(shortcut))
+
+    def _toggle_show_errors_on_map_filter(self) -> None:
```

Comment:
Checkboxes have a toggle method already. I think you could connect self.toggle_show_errors_on_map_action.triggered to that directly.

Reply:
Changed to use directly checkbox's toggle method